### PR TITLE
Clean Up Todos and Add in Issue Checker

### PR DESCRIPTION
**Pull request by nisdas**
This cleans up Todos referencing closed issues and adds in an issue checker to script that checks the status of issues

### CODE REVIEW

File: validator/proposer/service.go [hunk 1]
```diff
@@ -169,7 +169,6 @@ func (p *Proposer) run(done <-chan struct{}, client pb.ProposerServiceClient) {
 
 			bitmask := p.GenerateBitmask(p.pendingAttestation)
 
-			// TODO(#552): Implement real proposals with randao reveals and attestation fields.
```
[the remark this answers is not in the document]
Comment (nisdas) [reply validator/proposer/service.go]: The script basically just checks using the github api if the issue is closed. If it is closed it throws an error. I am just removing todos that are referencing issues that are closed

[the remark this answers is not in the document]
Comment (terenc3t) [reply validator/proposer/service.go]: Np. I can add a TODO there later.

**Comment by terenc3t** [comment]
Workflow wise - is it just me or this feels weird?

1. Bob wants to work on issue #1
2. Bob creates PR #2
3. Bob requests review and gets approved
4. Bob merges PR #2 
5. Bob forgets to delete the todo #1
6. Alice wants to work on issue #3 
7. Alice creates PR #4
8. Alice's build is failing because Bob forgets to delete his todo, now Alice has to find out what Bob did

It feels like the next person will be cleaning up for the last person quite often 😅 I feel everything should be contained within Bob.

File: validator/proposer/service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nisdas) [reply validator/proposer/service.go]: I added it back with #619.

**Comment by terenc3t** [comment]
Got it. We can give this a try. @prestonvanloon Do you have any thoughts for a tool like this?

File: beacon-chain/blockchain/service.go [hunk 2]
```diff
@@ -76,8 +76,6 @@ func NewChainService(ctx context.Context, cfg *Config) (*ChainService, error) {
 
 // Start a blockchain service's main event loop.
 func (c *ChainService) Start() {
-	// TODO(#474): Fetch the slot: (block, state) DAGs from persistent storage
```
[the remark this answers is not in the document]
Comment (rauljordan) [reply beacon-chain/blockchain/service.go]: I don't believe we need this as we already do everything using persistent storage. This seems outdated.

**Comment by rauljordan** [comment]
I think this should be fine for some bookkeeping but agreed that making it a merge requirement could lead into weird situations where we might be fixing other people's left over todo's. We can decide to make it a build item if others agree, but we can just merge this in for now as it isn't modifying travis.

File: scripts/check-todo.sh [hunk 3]
```diff
@@ -7,4 +7,18 @@ then
...
+issueNum=${line//$linenum}
+issueState=$(curl https://api.github.com/repos/prysmaticlabs/prysm/issues/$issueNum | grep -o '"state":"closed"');
+
+if [ "$issueState" != "" ];
+then 
+    echo "Issue referenced has already been closed" >&2;
+    echo "Issue Number: $issueNum" >&2;
+    exit 1;
 fi
+done < <(grep -PrinH -o -h '(?<!context\.)todo\(#{0,1}\K(\d+)' --include \*.go *)
```
[the remark this answers is not in the document]
Comment (nisdas) [reply scripts/check-todo.sh]: The do-while loop, reads any line printed out by grep, which parses the issue numbers from the files. Without that line the loop would not do anything as there is nothing printed out

[the remark this answers is not in the document]
Comment (prestonvanloon) [reply scripts/check-todo.sh]: Ah! Thanks!

**Pull request merged**